api/src/main/kotlin/edu/wgu/osmt/security/SecurityConfig.kt

```diff
@@ -68,42 +65,47 @@ class SecurityConfig : WebSecurityConfigurerAdapter() {
-            .antMatchers(HttpMethod.GET, scrubForConfigure(TASK_DETAIL_SKILLS)).authenticated()
-            .antMatchers(HttpMethod.GET, scrubForConfigure(TASK_DETAIL_BATCH)).authenticated()
-            .antMatchers(HttpMethod.GET, SEARCH_JOBCODES_PATH).authenticated()
-            .antMatchers(HttpMethod.GET, SEARCH_KEYWORDS_PATH).authenticated()
+            // authorization required
+            .antMatchers(HttpMethod.GET,  SKILLS_LIST).hasAnyAuthority(ADMIN, CURATOR, VIEW, READ)
+            .regexMatchers(HttpMethod.POST, createRegex(SKILL_UPDATE)).hasAnyAuthority(ADMIN, CURATOR)
+            .antMatchers(HttpMethod.POST, SKILLS_CREATE).hasAnyAuthority(ADMIN)
+            .antMatchers(HttpMethod.POST, SKILL_PUBLISH).hasAnyAuthority(ADMIN)
+
+            .antMatchers(HttpMethod.GET,  COLLECTIONS_LIST).hasAnyAuthority(ADMIN, CURATOR, VIEW, READ)
+            .regexMatchers(HttpMethod.POST, createRegex(COLLECTION_UPDATE)).hasAnyAuthority(ADMIN, CURATOR)
+            .antMatchers(HttpMethod.POST, COLLECTION_CREATE).hasAnyAuthority(ADMIN)
+            .antMatchers(HttpMethod.POST, COLLECTION_PUBLISH).hasAnyAuthority(ADMIN)
+            .regexMatchers(HttpMethod.POST, createRegex(COLLECTION_SKILLS_UPDATE)).hasAnyAuthority(ADMIN)
+
+            .regexMatchers(HttpMethod.POST, createRegex(COLLECTION_AUDIT_LOG)).permitAll()
+            .regexMatchers(HttpMethod.GET, createRegex(TASK_DETAIL_SKILLS)).permitAll()
+            .regexMatchers(HttpMethod.GET, createRegex(TASK_DETAIL_BATCH)).permitAll()
 
             // public search endpoints
-            .antMatchers(HttpMethod.GET,  SKILLS_LIST).permitAll()
             .antMatchers(HttpMethod.POST, SEARCH_SKILLS).permitAll()
-            .antMatchers(HttpMethod.GET,  COLLECTIONS_LIST).permitAll()
             .antMatchers(HttpMethod.POST, SEARCH_COLLECTIONS).permitAll()
 
             // public canonical URL endpoints
-            .antMatchers(HttpMethod.GET, scrubForConfigure(SKILL_DETAIL)).permitAll()
-            .antMatchers(HttpMethod.GET, scrubForConfigure(COLLECTION_DETAIL)).permitAll()
-            .antMatchers(HttpMethod.POST, scrubForConfigure(COLLECTION_SKILLS)).permitAll()
-            .antMatchers(HttpMethod.GET, scrubForConfigure(COLLECTION_CSV)).permitAll()
-            .antMatchers(HttpMethod.GET, scrubForConfigure(TASK_DETAIL_TEXT)).permitAll()   // public csv results
+            .regexMatchers(HttpMethod.GET, createRegex(SKILL_DETAIL)).permitAll()
+            .regexMatchers(HttpMethod.GET, createRegex(COLLECTION_DETAIL)).permitAll()
+
+            .regexMatchers(HttpMethod.POST, createRegex(COLLECTION_SKILLS)).permitAll()
+            .regexMatchers(HttpMethod.GET, createRegex(COLLECTION_CSV)).permitAll()
+            .regexMatchers(HttpMethod.GET, createRegex(TASK_DETAIL_TEXT)).permitAll()   // public csv results
 
             // catch-all
-            .antMatchers("/**").permitAll()
```

Comment:
This line may still be needed in the open source version. They server the static client assets from the Spring app, and I think this is required. Try running a local Spring app with no ng serve and not permitAll(), see if you still get the front end?

Reply:
We are able to run the UI on our local with no "ng serve". Need clarification/confirmation with JK